gap/PolygonalComplexes/animating.gd

```diff
@@ -0,0 +1,537 @@
+
+#! @Chapter Animating surfaces with Java Script
+#! @ChapterLabel AnimationJavaScript
+#!
+#! This chapter deals with animating simplicial surfaces via Java Script
+#! using three.js, see <URL>https://threejs.org/</URL>.
+
+#! @Section Introduction and Quick Start
+#! @SectionLabel LabelIntroductionAndQuickStartAnimating
+
+#! This section contains a minimal example by animating an octahedron and after that shows the general strategy to use the functions of this chapter.
+#! To construct an octahedron we only need to specify the 3D-coordinates of the vertices.
+#! @BeginExampleSession
+#! gap> oct := Octahedron();;
+#! gap> verticesPositions := [
+#! > [ 0, 0, Sqrt(2.) ],
+#! > [ 1, 1, 0 ],
+#! > [ 1, -1, 0 ],
+#! > [ -1, -1, 0 ],
+#! > [ -1, 1, 0 ],
+#! > [ 0, 0, -Sqrt(2.) ] ];;
+#! gap> printRecord := SetVertexCoordiantes3D(oct, verticesPositions, rec());;
+# rec( vertexCoordinates3D := [ [ 0, 0, 1.41421 ], [ 1, 1, 0 ], [ 1, -1, 0 ], [ -1, -1, 0 ], [ -1, 1, 0 ], [ 0, 0, -1.41421 ] ] )
+#! gap> DrawSurfaceToJavaScript(oct, "octahedron.html", printRecord);;
+#! @EndExampleSession
+#! Now your working directory should contain a file "octahedron.html". Below is shown an image of the animation.
+#! @InsertChunk Example_OctahedronAnimating
+
+#! The philosophy of the code presented in this chapter is the following.
+#! The main function <K>DrawSurfaceToJavaScript</K> (Section <Ref Sect="Section_LabelCoordinatesAndCoreFunctionality"/>) animates the surfaces using the configuration saved in the so called  print record, e.g. the 3D-coordinates (3D-coordinates are defined as a list with 3 entries of floats) of the vertices. It is necessary that the print record contains the 3D-coordinates of each vertex of the surface which should be animated. Based on the 3D-coordinates of the vertices the surface is placed in the animation.
+#! Therefore a typical workflow to animate a surfaces looks like this:
+#! <Enum>
+#!   <Item>Construct the surface, e.g. the octahedron above.</Item>
+#!   <Item>Configure your animation using the functions in the Sections
+#!   <Ref Subsect="Section_LabelVisibility"/> and <Ref Subsect="Section_LabelColouring"/>.</Item>
+#!   <Item>Write your animation to a file calling the main function <K>DrawSurfaceToJavaScript</K> (see Section <Ref Sect="Section_LabelCoordinatesAndCoreFunctionality"/>).</Item>
+#! </Enum>
+#! The output of the function <K>DrawSurfaceToJavaScript</K> is a html file. The file can be opened with any browser and then the animation is shown directly.
+
+#! @Section Coordinates and Core Functionality
+#! @SectionLabel LabelCoordinatesAndCoreFunctionality
+
+#! This section deals with the main functions. Before the animation (see <K>DrawSurfaceToJavaScript</K>) can be drawn, it is necessary to specify the 3D-coordinates (which means, the [x,y,z] coordinates) of all vertices of the surface which should be animated (see <K>SetVertexCoordinates3D</K>).
```

Comment:
This section describes the animation method and how to define vertex coordinates. ...

If you refer to methods, you should also set a link (otherwise they can be cumbersome to find).